src/vs/workbench/contrib/terminal/node/terminalProcess.ts

```diff
@@ -69,16 +70,48 @@ export class TerminalProcess implements ITerminalChildProcess, IDisposable {
 			experimentalUseConpty: useConpty
 		};
 
-		// TODO: Need to verify whether executable is on $PATH, otherwise things like cmd.exe will break
-		// fs.stat(shellLaunchConfig.executable!, (err) => {
-		// 	if (err && err.code === 'ENOENT') {
-		// 		this._exitCode = SHELL_PATH_INVALID_EXIT_CODE;
-		// 		this._queueProcessExit();
-		// 		this._processStartupComplete = Promise.resolve(undefined);
-		// 		return;
-		// 	}
-		this.setupPtyProcess(shellLaunchConfig, options);
-		// });
+		fs.stat(shellLaunchConfig.executable!, (err) => {
+			if (err && err.code === 'ENOENT' && !this.checkIfExistsInPath(shellLaunchConfig.executable!)) {
```

Comment:
This doesn't seem to work either:

```
"terminal.integrated.shell.osx": "bash",
```

<img width="452" alt="Screen Shot 2019-06-28 at 11 26 20 AM" src="https://user-images.githubusercontent.com/2193314/60363159-94f97a80-9997-11e9-9e28-f3fca0bd56ba.png">